CONTRIBUTING.md

```diff
@@ -1,86 +1,187 @@
 
-We also appreciate it if you take the time to update and write tests for any changes
-you submit.
+Contributions are very welcome. If you think you need help planning your contribution, please ping us on [Slack](https://slack.pact.io) and let us know you are looking for a bit of help.
 
-[examples]: https://github.com/pact-foundation/pact-python/tree/master/examples
+### Join our Slack
 
-## Commit messages
+We have a [Slack](https://slack.pact.io) to discuss all things about Pact and its development. Feel free to ask questions about Pact Python specifically in the [`#pact-python`](https://pact-foundation.slack.com/archives/C9VECUP6E) channel, or broader questions about the Pact ecosystem over in the [`#general`](https://pact-foundation.slack.com/archives/C5F4KFKR8) channel.
 
-pact-python is adopting the [Conventional Commits](https://www.conventionalcommits.org)
-convention. Please ensure you follow the guidelines, we don't want to be that person,
-but the commit messages are very important to the automation of our release process.
+### Triaging Issues and Pull Requests
 
-Take a look at the git history (`git log`) to get the gist of it.
+One great way you can contribute to the project without writing any code is to help triage issues and pull requests as they come in.
 
-If you'd like to get some CLI assistance there is a node npm package. Example usage is:
+-   Ask for more information if you believe the issue does not provide all the details required to solve it.
+-   Suggest [labels](https://github.com/pact-foundation/pact-python/labels) that can help categorize issues.
+-   Flag issues that are stale or that should be closed.
+-   Ask for test plans and review code.
 
-```shell
-npm install -g commitizen
-npm install -g cz-conventional-changelog
-echo '{ "path": "cz-conventional-changelog" }' > ~/.czrc
-```
+## Our Development Process
 
-When you commit with Commitizen, you'll be prompted to fill out any required
-commit fields at commit time. Simply use `git cz` or just `cz` instead of
-`git commit` when committing. You can also use `git-cz`, which is an alias
-for `cz`.
+Pact Python uses [GitHub](https://github.com/pact-foundation/pact-python) as its source of truth. The team will be working directly there. All changes will be public from the beginning.
 
-See https://www.npmjs.com/package/commitizen for more info.
+All pull requests will be checked by the continuous integration system, GitHub actions. There are unit tests, end-to-end tests, performance tests, style tests, and much more.
 
-There is a pypi package that does similar [commitizen](https://pypi.org/project/commitizen/).
+### Branch Organization
 
-## Running the tests
+Pact Python has one primary branch `master` and we use feature branches to deliver new features with pull requests. We use the following naming convention for branches:
```

Comment:
We probably want to have the conversation about master/main and implications across the estate.

I'm okay with it being master, but appreciate others reservations. My only consideration with regards to programming is for consistency.

New repos are created with main by default, so at some point we should probably migrate to main across older repos.

We call it a main branch detection property in the pact broker

https://docs.pact.io/pact_broker/branches#automatic-main-branch-detection

>To assist in the migration from tags to branches, the main branch will be automatically set for a pacticipant if a version is created with a branch or tag name matching one of develop, main, or master.

so from a code perspective, we support both, so its just a question on if and when as maintainers, we want to make the switch.

Bit of an aside from here, but worth calling it.

Another note, it is a protected branch, and cannot be merged to without at least one approver, the same goes fo for the release process, although that is documented seperately

Reply:
I completely agree that we should migrate to `main`. Most platforms have migrated to having `main` as the default, and I believe Git even has that as the default now.

I believe there's also a distinction between casually referring to the main branch vs the `main` branch. In the former, this could be anything (e.g., `develop` or `stable`, or it could be `v2-main`), whereas in the latter, we are explicitly referring to the branch called `main`. Admittedly, this is a bit of a nit-pick.

The problem is that changing from `master` to `main` will break things for people, and I would rather not do this right now.

At this stage, I have documented what we are currently using, and if/when we migrate `master` to `main`, the docs can be updated.

As for the merging process, I think that's fine. It's pretty self-explanatory once the PR is created.